File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int n = nums.size();
        if(n == 1) return nums[0];

        for(int i=0; i<=n-2; i++){
            if(i == 0){
                if(nums[i] != nums[i+1])
                    return nums[i];
            }
            else if(nums[i-1] != nums[i] && nums[i+1] != nums[i]){
                return nums[i];
            }
        }
        return nums[n-1];
    }
};

File: 540-single-element-in-a-sorted-array/single-element-in-a-sorted-array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int n = nums.length;
        if(n == 1) return nums[0];
        for(int i=1; i<nums.length-1; i++){
            if(nums[i] == nums[i-1] || nums[i] == nums[i+1]){
                continue;
            }else return nums[i];
        }
        if(nums[0] != nums[1]) return nums[0];
        return nums[n-1];
    }
}